### PR TITLE
sssp implementation in charm4py

### PR DESCRIPTION
**Pull request by ritvikrao**
I added a single-source shortest paths (SSSP) example in Charm4Py. You provide a number of vertices, a number of edges, a random seed, and a source vertex. A graph is automatically generated, and then a parallel Bellman-Ford SSSP algorithm is executed on that graph.

### CODE REVIEW

**Comment by lvkale** [comment]
Have you tested at some reasonable scale (for both graph size and number of pes)?

**Comment by ritvikrao** [comment]
> Have you tested at some reasonable scale (for both graph size and number of pes)?

I have so far run up to 2M edges on 48 PEs.

File: examples/sssp/sssp.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            edge_dest = random.randint(0, self.num_vertices-1)
+            while edge_source==edge_dest:
+                edge_dest = random.randint(0, self.num_vertices-1)
+            edge_weight = random.random()
+            self.edge_list.append((edge_source, edge_dest, edge_weight))
+        self.edge_list.sort(key=lambda a: a[0])
+        #initiate worker array
+        num_partitions = charm.numPes()
+        self.workers = Array(SsspChares, num_partitions)
+        charm.awaitCreation(self.workers)
```
Comment (adityapb) [inline examples/sssp/sssp.py:78]: Why is this call necessary?

Comment (ritvikrao) [reply examples/sssp/sssp.py:78]: From what I saw in other examples, I thought this was necessary to start calling entry methods on the chare array. Some testing now shows that it's not needed, so I'll delete it.

**Review by lvkale — APPROVED** [review]

**Pull request merged**